### PR TITLE
Improve proxy saftey & logging, update useSearch to support jobs endpoint

### PR DESCRIPTION
**Pull request by dauglyon**
See self-review below for a summary of changes

### CODE REVIEW

File: react_app/src/components/SearchResults.tsx [hunk 1]
```diff
@@ -68,22 +68,25 @@ export const SearchResults: FC<SearchResultsProps> = ({ searches, addSearchById,
...
+                )}
+                title={<span>Search ID: <Typography.Text code>{search.id}</Typography.Text></span>}
+                description={<span>Status: {search.status ?? 'unknown'}</span>} />
+              <ResultContent search={search} />
+            </List.Item>
+          )} />
+      </Typography.Paragraph>
     </div>
   );
 };
```
Comment (dauglyon) [inline react_app/src/components/SearchResults.tsx:92]: Changes here are just formatting

File: react_app/src/hooks/useSearches.tsx [hunk 2]
```diff
@@ -3,13 +3,14 @@ import { notification } from 'antd';
 import { useLocalStorage } from '@rehooks/local-storage';
 import { Search } from '../types/Search';
 import { Namespace } from '../types/Namespace';
-import { usePrevious } from './usePrevious';
 import { urlFor } from '../utils/urlFor';
 import { parseFASTA } from '../utils/handleFASTA';
 
-async function fetchStatuses() {
+async function fetchStatuses(ids: Search['id'][]) {
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:9]: Added support for the new `jobIds` required API param

File: react_app/src/hooks/useSearches.tsx [hunk 3]
```diff
@@ -53,34 +54,37 @@ export function useSearch() {
...
 
     const poll = async () => {
-      const next = await fetchStatuses();
+      const next = await fetchStatuses(pending);
       setSearchStatus(prev => ({ ...prev, ...next }))
     };
 
     poll();
-    const pollInterval = setInterval(poll, 3000);
+    const pollInterval = setInterval(poll, 15000);
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:70]: Increased polling interval

File: react_app/src/hooks/useSearches.tsx [hunk 4]
```diff
@@ -53,34 +54,37 @@ export function useSearch() {
   const [searchResults, setSearchResults] = useState<Record<Search['id'], Search['result']>>({});
 
   // Poll for search status changes
-  const shouldPoll = searchIds.some(id => searchStatus[id] !== "completed");
+  const pendingSerialized = JSON.stringify(
+    searchIds.filter(id => searchStatus[id] !== "completed")
+  );
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:59]: Proper `useEffect` dependencies must be `==` comparable for identity. Serializing to a JSON string works well here (and below for `completedSerialized`) and is more readable IMO than then more complicated `useRef` hack.

File: react_app/src/hooks/useSearches.tsx [hunk 5]
```diff
@@ -95,7 +99,8 @@ export function useSearch() {
       });
       if (response.ok) {
         const search: Search = await response.json();
-        addSearchById(search.id, search.status);
+        setSearchIds([search.id, ...searchIds])
+        setSearchStatus(prev => ({ ...prev, [search.id]: search.status }))
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:103]: removed call to `addSearchById` to skip the new existence check.

File: react_app/src/hooks/useSearches.tsx [hunk 6]
```diff
@@ -108,9 +113,26 @@ export function useSearch() {
     }
   }
 
-  function addSearchById(id: Search['id'], status: Search['status'] = undefined) {
-    setSearchIds([id, ...searchIds])
-    setSearchStatus(prev => ({ ...prev, id: status }))
+  async function addSearchById(id: Search['id']) {
```
Comment (dauglyon) [inline react_app/src/hooks/useSearches.tsx:116]: Added a call to the API checking the ID exists so as to not cause issues downstream.

File: react_app/src/utils/urlFor.ts [hunk 7]
```diff
@@ -1,4 +1,8 @@
-export function urlFor(path: string) {
-  if (path.startsWith("/")) return process.env.PUBLIC_URL + path;
-  else return `${document.URL}${document.URL.endsWith("/") ? "" : "/"}${path}`;
+export function urlFor(path: string, params?: Record<string, string>) {
```
Comment (dauglyon) [inline react_app/src/utils/urlFor.ts:1]: Added url param formatting

File: sequence_search/config.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import os
+from .util import valid_url
+
+
+def genesearch_url():
```
Comment (dauglyon) [inline sequence_search/config.py:5]: Checks that `os.environ['GENESEARCH_URL']` is valid before starting the server

File: sequence_search/app.py [hunk 9]
```diff
@@ -1,19 +1,42 @@
...
+from .config import GENESEARCH_URL
+from .util import valid_url
+
 app = Sanic("sequence_search")
 
 
 @app.route('/api/v1/<api_path:path>', methods=["GET", "POST"])
 async def api_proxy(request, api_path):
-    api_url = urllib.parse.urljoin(os.environ['GENESEARCH_URL'], api_path)
+    """
```
Comment (dauglyon) [inline sequence_search/app.py:16]: Added some saftey checks and url param string forwarding

Comment (dauglyon) [reply sequence_search/app.py:16]: Also now logs the original url _and_ proxy target.

File: react_app/src/hooks/useSearches.tsx (hunk 2 above)
Comment (dauglyon) [reply react_app/src/hooks/useSearches.tsx:9]: see https://lb.genesearch.production.svc.spin.nersc.org/docs/#operation/getJobStatuses

**Review by dakotablair — APPROVED** [review]
Looks good to me! The logging is quite clear and does not seem overly verbose.

**Pull request merged**